euphonic/force_constants.py

```diff
@@ -556,8 +556,7 @@ def _calculate_phonons_at_qpts(
         # expensive phase calculations later
         sc_image_r = get_all_origins(
             np.repeat(n_sc_shells, 3) + 1, min_xyz=-np.repeat(n_sc_shells, 3))
-        sc_origins = np.einsum('ij,jk->ik', sc_image_r,
-                               self.sc_matrix).astype(np.int32)
+        sc_origins = (sc_image_r @ self.sc_matrix).astype(np.int32)
         unique_sc_origins = [[] for i in range(3)]

```

Comment:
Any reason why this is the empty list when it's overwritten? Would

`[None]*3` not be clearer as to intent?

Reply:
I think None would be less clear because it makes it look more like a flat 3-vector! It is a `list[list[int]]` and we don't know in advance what the list lengths will be except that the first dimension is 3.